src/app/backend/resource/deployment/deploy.go

```diff
@@ -285,55 +290,68 @@ func getLabelsMap(labels []Label) map[string]string {
-func DeployAppFromFile(cfg clientcmd.ClientConfig, spec *AppDeploymentFromFileSpec,
-	createObjectFromInfoFn createObjectFromInfo) (bool, error) {
-	const emptyCacheDir = ""
-	validate := spec.Validate
+func DeployAppFromFile(cfg *rest.Config, spec *AppDeploymentFromFileSpec) (bool, error) {
+	reader := strings.NewReader(spec.Content)
+	log.Printf("Namespace for deploy from file: %s\n", spec.Namespace)
+	d := yaml.NewYAMLOrJSONDecoder(reader, 4096)
+	for {
+		data := unstructured.Unstructured{}
+		if err := d.Decode(&data); err != nil {
+			if err == io.EOF {
+				return true, nil
+			}
+			return false, err
+		}
 
-	factory := cmdutil.NewFactory(cfg)
-	schema, err := factory.Validator(validate, emptyCacheDir)
-	if err != nil {
-		return false, err
-	}
+		version := data.GetAPIVersion()
+		kind := data.GetKind()
 
-	mapper, typer := factory.Object()
-	reader := strings.NewReader(spec.Content)
+		gv, err := schema.ParseGroupVersion(version)
+		if err != nil {
+			gv = schema.GroupVersion{Version: version}
+		}
 
-	fmt.Printf("Namespace for deploy from file: %s\n", spec.Namespace)
+		groupVersionKind := schema.GroupVersionKind{Group: gv.Group, Version: gv.Version, Kind: kind}
 
-	builder := kubectlResource.NewBuilder(mapper, kubectlResource.LegacyCategoryExpander, typer,
-		kubectlResource.ClientMapperFunc(factory.ClientForMapping), factory.Decoder(true)).
-		Schema(schema).
-		NamespaceParam(spec.Namespace).
-		Stream(reader, spec.Name).
-		Flatten()
+		discoveryClient, err := discovery.NewDiscoveryClientForConfig(cfg)
+		if err != nil {
+			return false, err
+		}
 
-	if strings.Compare(spec.Namespace, "_all") != 0 {
-		builder.RequireNamespace()
-	}
+		apiResourceList, err := discoveryClient.ServerResourcesForGroupVersion(version)
+		if err != nil {
+			return false, err
+		}
+		apiResources := apiResourceList.APIResources
+		var resource *metaV1.APIResource
+		for _, apiResource := range apiResources {
+			if apiResource.Kind == kind && !strings.Contains(apiResource.Name, "/") {
+				resource = &apiResource
+				break
+			}
+		}
+		if resource == nil {
+			return false, fmt.Errorf("Unknown resource kind: %s", kind)
+		}
 
-	r := builder.Do()
+		dynamicClientPool := dynamicclient.NewDynamicClientPool(cfg)
 
-	deployedResourcesCount := 0
+		dynamicClient, err := dynamicClientPool.ClientForGroupVersionKind(groupVersionKind)
 
-	err = r.Visit(func(info *kubectlResource.Info, err error) error {
-		isDeployed, err := createObjectFromInfoFn(info)
-		if isDeployed {
-			deployedResourcesCount++
-			log.Printf("%s is deployed", info.Name)
+		if err != nil {
+			return false, err
 		}
-		return err
-	})
 
-	err = errors.LocalizeError(err)
-	return deployedResourcesCount > 0, err
+		if strings.Compare(spec.Namespace, "_all") == 0 {
+			_, err = dynamicClient.Resource(resource, data.GetNamespace()).Create(&data)
+		} else {
+			_, err = dynamicClient.Resource(resource, spec.Namespace).Create(&data)
+		}
+
+		if err != nil {
+			return false, err
```

Comment:
`return false, errors.LocalizeError(err)` here as it was done before. We need that to correctly localize error in frontend.